src/concurrency/thread.rs

```diff
@@ -32,9 +32,9 @@ pub enum SchedulingAction {
 
 /// Timeout callbacks can be created by synchronization primitives to tell the
 /// scheduler that they should be called once some period of time passes.
-type TimeoutCallback<'mir, 'tcx> = Box<
-    dyn FnOnce(&mut InterpCx<'mir, 'tcx, MiriMachine<'mir, 'tcx>>) -> InterpResult<'tcx> + 'tcx,
->;
+pub trait TimeoutCallback<'mir, 'tcx>: VisitMachineValues + 'tcx {
+    fn call(&self, ecx: &mut InterpCx<'mir, 'tcx, MiriMachine<'mir, 'tcx>>) -> InterpResult<'tcx>;
+}
```

Comment:
You need to remove the `+'tcx` from the super trait list and add it only to the `Box<dyn Trait + 'tcx>`, otherwise you get an implication that `'mir: 'tcx` due to the `'mir` on the trait

Reply:
Hunh. I'm sure that will seem intuitive to me later.